streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1644,6 +1668,29 @@ private Map<String, Object> getClientCustomProps() {
         return props;
     }
 
+    public static List<String> verifyTopologyOptimizationConfigs(final String config) {
+        final List<String> acceptableConfigs = Arrays.asList(
```

Comment:
I wonder if this list of optimization flags should be a static variable somewhere?  

Maybe it is fine if it is only used here.